torch/jit/__init__.py

```diff
@@ -656,6 +656,25 @@ def __getitem__(self, k):
             raise KeyError(k)
         return self.module._get_parameter(k)
 
+# types that can be constants. If you edit this list,
+# then you also need to edit the handlers in ConstantValue in jit/script/init.cpp
+_constant_types = [bool, float, int, Module, types.FunctionType]
```

Comment:
Do you need to mark `nn.Module` as constant to be able to use it inside scripted functions? Also, it doesn't really mean that the module (as in its state, attributes, etc.) will remain constant, right? Only that this attribute won't be re-assigned later

Reply:
No you don't need to mark a module const, but you do want to label a _list_ of modules const. Maybe this should be more integrated with ModuleList, I will look into how it works.